server/events/command_runner.go

```diff
@@ -134,55 +134,89 @@ func (c *DefaultCommandRunner) RunAutoplanCommand(baseRepo models.Repo, headRepo
 		return
 	}
 
-	projectCmds, err := c.ProjectCommandBuilder.BuildAutoplanCommands(ctx)
+	// Run plan command for all projects
+	planErr := c.runAutoCommand(ctx, models.PlanCommand, AutoplanCommand{})
+
+	if planErr == nil {
+		// Run policy_check command
+		_ = c.runAutoCommand(ctx, models.PolicyCheckCommand, AutoPolicyCheckCommand{})
+	}
+}
+
+func (c *DefaultCommandRunner) runAutoCommand(ctx *CommandContext, cmdModel models.CommandName, pullCommand PullCommand) error {
+	var projectCmds []models.ProjectCommandContext
+	var err error
+
+	switch cmdModel.String() {
+	case "plan":
+		projectCmds, err = c.ProjectCommandBuilder.BuildAutoplanCommands(ctx)
+	case "policy_check":
+		projectCmds, err = c.ProjectCommandBuilder.BuildAutoPolicyCheckCommands(ctx)
+	}
+
 	if err != nil {
-		if statusErr := c.CommitStatusUpdater.UpdateCombined(ctx.Pull.BaseRepo, ctx.Pull, models.FailedCommitStatus, models.PlanCommand); statusErr != nil {
+		if statusErr := c.CommitStatusUpdater.UpdateCombined(ctx.Pull.BaseRepo, ctx.Pull, models.FailedCommitStatus, cmdModel); statusErr != nil {
 			ctx.Log.Warn("unable to update commit status: %s", statusErr)
 		}
 
-		c.updatePull(ctx, AutoplanCommand{}, CommandResult{Error: err})
-		return
+		c.updatePull(ctx, pullCommand, CommandResult{Error: err})
+		return err
 	}
+
 	if len(projectCmds) == 0 {
-		log.Info("determined there was no project to run plan in")
+		ctx.Log.Info("determined there was no project to run %s in", cmdModel.String())
 		if !c.SilenceVCSStatusNoPlans {
 			// If there were no projects modified, we set a successful commit status
 			// with 0/0 projects planned successfully because some users require
 			// the Atlantis status to be passing for all pull requests.
 			ctx.Log.Debug("setting VCS status to success with no projects found")
-			if err := c.CommitStatusUpdater.UpdateCombinedCount(baseRepo, pull, models.SuccessCommitStatus, models.PlanCommand, 0, 0); err != nil {
+			if err := c.CommitStatusUpdater.UpdateCombinedCount(ctx.Pull.BaseRepo, ctx.Pull, models.SuccessCommitStatus, cmdModel, 0, 0); err != nil {
 				ctx.Log.Warn("unable to update commit status: %s", err)
 			}
 		}
-		return
+		return err
 	}
```

Comment:
Presumably, this would already be resolved by the policy checking stage so it's unfortunate we still have to do these kinds of checks given the way commands are built.  Given that policy checking always follows plan, did you consider just adding it as a step in the plan command?  I'm unsure of how complex it is to deal with 2 sets of outputs and 2 commit statuses but the benefits I see are:

* Less duplicated logic to run through (ie. all the project config stuff)
* Less code to write here
* You get parallel flag for free and don't need to create one for policy checking